src/densities.jl

```diff
@@ -118,18 +111,15 @@ end
                      δoccupation[ik][n] .* abs2.(ψnk_real))
         end
         δρk_fourier = r_to_G(basis, complex(δρk))
-        lowpass_for_symmetry!(δρk_fourier, basis)
         accumulate_over_symmetries!(δρ_fourier[:, :, :, kpt.spin], δρk_fourier,
                                     basis, basis.ksymops[ik])
     end
     mpi_sum!(δρ_fourier, basis.comm_kpts)
     count = sum(length(basis.ksymops[ik]) for ik in 1:length(basis.kpoints)) ÷ n_spin
     count = mpi_sum(count, basis.comm_kpts)
+    lowpass_for_symmetry!(δρ_fourier, basis)
     δρ = G_to_r(basis, δρ_fourier) ./ count
-    # Check sanity in the density, we should have ∫δρ = 0
-    if abs(sum(δρ)) > sqrt(eps(T))
-        @warn "Non-neutral δρ" sum_δρ=sum(δρ)
-    end
+    _check_total_charge(basis.dvol, δρ, 0)
```

Comment:
I still think this will yield to a confusing error message. People will run response or AD when this pops up and then reading something about the number of electrons being 0 is clearly not what they expect.

Reply:
It's not really feasible long-term to have different error messages for the AD and not AD version. Simplest is just to turn off the check?

Reply:
Actually I agree. This is probably not very useful to have here.